software/fasttree/main.nf

```diff
@@ -0,0 +1,40 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process FASTTREE {
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:'') }
+
+    conda (params.enable_conda ? "bioconda::fasttree=2.1.10" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fasttree:2.1.10--h516909a_4"
+    } else {
+        container "quay.io/biocontainers/fasttree:2.1.10--h516909a_4"
+    }
+
+    input:
+    path alignment
+
+    output:
+    path "*.tre",         emit: phylogeny
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    """
+    fasttree \\
+        -gtr \\
+        -gamma \\
+        -fastest \\
+        -log fasttree_phylogeny.tre.log \\
+        -nt $alignment \\
+        > fasttree_phylogeny.tre
+
+    fasttree -help 2>&1 | head -1  | sed 's/^FastTree \\([0-9\\.]*\\) .*\$/\\1/' > ${software}.version.txt
```

Comment:
Wrap this first bit in an `echo` which effectively cancels out weird exit codes generated by some tools when running `-h` or `-v` e.g. 
https://github.com/nf-core/modules/blob/710ab76b90f23f288c2d60400dc162149ff5e206/software/samtools/faidx/main.nf#L32